### PR TITLE
Per capita review

### PR DESCRIPTION
**Pull request by Stephen-ONeil**
Scatter comments throughout to add the same descriptive feedback I would normally provide in a standard review process. 

Tried to keep commits focused and well explained in commit messages too, although some grew out of scope and a number made changes I later re-thought. Still, might be useful to at least scan  through commit-by-commit, if you want to follow my process.

TODO:
  - [x] coordinate with pipeline on csv name change

### CODE REVIEW

**Comment by Stephen-ONeil** [comment]
Note: I started some cleanup around the changes made to the `subj_name` handlebars helper on the show-per-capita branch, but my changes got out of scope (and touched too many scattered lines for me to want it cluttering up this branch). Shifted those changes over to #562

File: client/src/components/TabbedContent.js [hunk 1]
```diff
@@ -15,23 +15,24 @@ export class TabbedControls extends React.Component {
...
-                  className = "tabbed-controls__label-text"
-                  role = "button"
-                  aria-pressed = { is_open }
-                  onClick = { () => !is_disabled && tab_callback(key) }
-                  onKeyDown = { (e) => !is_disabled && (e.keyCode===13 || e.keyCode===32) && tab_callback(key) }
+                  tabIndex={0}
+                  className="tabbed-controls__label-text"
+                  role="button"
+                  aria-pressed={ is_open }
+                  aria-disabled={ is_disabled }
```
Comment (Stephen-ONeil) [inline client/src/components/TabbedContent.js]: Hard to see because of all the code style tweaks, but the only a11y changes I made to `TabbedContent` were properly flagging disabled state to screen readers here, and removing the `aria-hidden` below.

File: client/src/components/TabbedContent.js [hunk 2]
```diff
@@ -81,23 +82,26 @@ export class TabbedContent extends React.Component {
         key,
         label: tab_labels[key],
         is_open: open_tab_key === key,
-        is_disabled: (tab_is_disabled)?tab_is_disabled[key]:false,
+        is_disabled: _.includes(disabled_tabs, key),
```
Comment (Stephen-ONeil) [inline client/src/components/TabbedContent.js]: Since having no disabled tabs is the default state, wanted the `disabled_tabs` prop to be totally optional. This tweak, and the default empty array value set below, do that.

File: client/src/panels/panel_declarations/InfographicPanel.js [hunk 3]
```diff
@@ -174,16 +174,21 @@ Col.propTypes = {
...
+  .thru( (children) => _.isArray(children) ? children : [children] )
+  .flatMap( child => {
+    if ( child.type && child.type === Fragment ){
+      return child.props.children;
+    } else {
+      return child;
+    }
+  })
+  .filter( child => child && child.type )
+  .value();
```
Comment (Stephen-ONeil) [inline client/src/panels/panel_declarations/InfographicPanel.js]: This was an unrelated diversion, ran in to this really frustrating logic bug in the `StdPanel` component while trying out different layouts for the tp_by_region panel.

File: client/src/panels/panel_declarations/people/employee_age.js [hunk 4]
```diff
@@ -166,28 +166,30 @@ export const declare_employee_age_panel = () => declare_panel({
...
-                  avgage: (
-                    <div id={"emp_age_tab_pane"}>
-                      <LineBarToggleGraph {...avg_age_options} />
-                      <div className='clearfix'></div>
-                    </div>
-                  ),
-                }}
-              />
+            <Col size={12} isGraph extraClasses="zero-padding">
+              <div aria-hidden="true">
```
Comment (Stephen-ONeil) [inline client/src/panels/panel_declarations/people/employee_age.js]: This is where the `aria-hidden="true"` from `TabbedContent` actually belonged.

File: client/src/panels/panel_declarations/finances/transfer_payments/tp_by_region.yaml [hunk 5]
```diff
@@ -6,39 +6,29 @@ tp_by_region_title:
...
-show_tp:
-  en: Transfer Payments
-  fr: Les Paiements de Transfert
 
-show_tp_per_capita:
-  en: Tranfer Payments per Capita
-  fr: Les Paiements de Transfert par Habitant
+tp_by_region_graph_usage:
+  en: Hover over or tap a province or territory in the map below to see the total transfer payments over the last five years. Note that the scale may change to accommodate the data displayed.
+  fr: Survolez ou tapez sur une province ou un territoire dans la carte ci-dessous pour voir le total des paiements de transfert au cours des cinq dernières années. Notez que l'échelle peut changer pour s'adapter aux données affichées.
```
Comment (Stephen-ONeil) [inline client/src/panels/panel_declarations/finances/transfer_payments/tp_by_region.yaml]: Split this out so it'd be easier to hide in the a11y mode, where it refers to a graph that doesn't exist.

File: client/src/panels/panel_declarations/finances/transfer_payments/tp_by_region.yaml [hunk 6]
```diff
@@ -6,39 +6,29 @@ tp_by_region_title:
...
+transfer_payments:
+  en: Transfer Payments
+  fr: Paiements de transfert
 
-tp_pc_a11y_table_title:
-  en: Transfer Payments for Provinces per Capita
-  fr: Paiements de Transfert pour les Provinces par Habitant
+transfer_payments_per_capita:
+  en: Transfer Payments per Capita
+  fr: Paiements de transfert par habitant
```
Comment (Stephen-ONeil) [inline client/src/panels/panel_declarations/finances/transfer_payments/tp_by_region.yaml]: Technically the graph's 'regions', not provinces (provinces and territories, and also whatever category you'd put "Abroad" and "Not Available" in). Simpler to just not specify.

File: client/src/panels/panel_declarations/finances/transfer_payments/tp_by_region.js [hunk 7]
```diff
@@ -65,183 +159,97 @@ class TPMap extends React.Component {
...
+      
+      const text_args = get_text_args(subject, transfer_payment_data, per_capita_data);
 
       return (
         <StdPanel
           title={text_maker("tp_by_region_title")}
           {...{ footnotes, sources }}
         >
           <Col size={12} isText>
+            <TM k="tp_by_region_text" args={text_args} />
```
Comment (Stephen-ONeil) [inline client/src/panels/panel_declarations/finances/transfer_payments/tp_by_region.js]: Moved the panel text above the tabs. I found that even I couldn't be bothered to read the text carefully enough to tell what was changing in it when the tabs switched. Becomes sort of a word search when you know the text is changing in response to a click, but it's only changing by about 15%. Simplified the design AND the code to just have one static set of text, existing outside of the control of the tabs.

**Comment by Stephen-ONeil** [comment]
In general, gave a lot of variables more descriptive/meaningful/consistent names. Also sectioned off as much logic as I thought reasonable from the render in to more tightly scoped functions.

The smaller functions do mean more hopping up and down the file while reading, but it gives clear names to the steps involved in preparing the data for rendering, and means that readers can focus on smaller blocks of code with less variables in scope at once/more explicit scopes (along with avoiding mutations, doing this makes it way easier to correctly read code without needing to step through with a debugger to double check your understanding of a given variable's contents).

**Comment by Stephen-ONeil** [comment]
@bill-lliu what are the good edge case departments to test this panel on?

I've already realized that the formatting on the per capita values doesn't work great when the per capita values can end up << 1, eg: http://dev.ea-ad.ca/per-capita-review/index-eng.html#orgs/dept/552/infograph/financial/.-.-(panel_key.-.-'tp_by_region)

**Comment by Stephen-ONeil** [comment]
TODO:
  - [ ] coordinate with pipeline on csv name change

**Comment by bill-lliu** [comment]
> @bill-lliu what are the good edge case departments to test this panel on?
> 
> I've already realized that the formatting on the per capita values doesn't work great when the per capita values can end up << 1, eg: http://dev.ea-ad.ca/per-capita-review/index-eng.html#orgs/dept/552/infograph/financial/.-.-(panel_key.-.-'tp_by_region)

Here are my Edge Cases @Stephen-ONeil Turns out the last one seems to be running into some issues; 
I spent a bit of time looking into it and it seems to be the same issue as before: there's data for only the Abroad category or the NA category, so the normal graph generates, but the show-per-capita graph cant generate anything and breaks somewhere
I kind of hard-coded some solutions to this but I think your review undid my hot-fixes
I can either re-do them or just leave it for now, but it comes up in "Infographic for Office of the Public Sector Integrity Commissioner"

Government
https://dev.ea-ad.ca/per-capita-review/index-eng.html#orgs/gov/gov/infograph/financial
Department
https://dev.ea-ad.ca/per-capita-review/index-eng.html#orgs/dept/1/infograph/financial
Missing Provinces
https://dev.ea-ad.ca/per-capita-review/index-eng.html#orgs/dept/326/infograph/financial
No Provinces
https://dev.ea-ad.ca/per-capita-review/index-eng.html#orgs/dept/250/infograph/financial

File: client/src/components/TabbedContent.js (hunk 1 above)
Comment (bill-lliu) [reply client/src/components/TabbedContent.js]: Accessible rich internet applications: for improving accessibility (visibility for people using screen readers)
👍

File: client/src/components/TabbedContent.js (hunk 2 above)
Comment (bill-lliu) [reply client/src/components/TabbedContent.js]: Logic makes sense
👍

File: client/src/panels/panel_declarations/InfographicPanel.js (hunk 3 above)
Comment (bill-lliu) [reply client/src/panels/panel_declarations/InfographicPanel.js]: Logic makes sense
👍

File: client/src/panels/panel_declarations/people/employee_age.js (hunk 4 above)
Comment (bill-lliu) [reply client/src/panels/panel_declarations/people/employee_age.js]: hides tabbed content correctly, makes sense
👍

File: client/src/panels/panel_declarations/finances/transfer_payments/tp_by_region.yaml (hunk 5 above)
Comment (bill-lliu) [reply client/src/panels/panel_declarations/finances/transfer_payments/tp_by_region.yaml]: Split yaml text for clarity
👍

File: client/src/panels/panel_declarations/finances/transfer_payments/tp_by_region.yaml (hunk 6 above)
Comment (bill-lliu) [reply client/src/panels/panel_declarations/finances/transfer_payments/tp_by_region.yaml]: Wording fix for yaml, makes sense
👍

File: client/src/panels/panel_declarations/finances/transfer_payments/tp_by_region.js (hunk 7 above)
Comment (bill-lliu) [reply client/src/panels/panel_declarations/finances/transfer_payments/tp_by_region.js]: Args were a little redundant, simplifying reduced clutter
👍
